orangecontrib/spectroscopy/widgets/owfft.py

```diff
@@ -532,35 +559,89 @@ def determine_sweeps(self):
 
     def check_metadata(self):
         """ Look for laser wavenumber and sampling interval metadata """
+       
+        # if self.data.attributes is not None:
         try:
-            lwn, _ = self.data.get_column_view("Effective Laser Wavenumber")
-        except ValueError:
-            if not self.dx_HeNe_cb.isEnabled():
-                # Only reset if disabled by this code, otherwise leave alone
-                self.dx_HeNe_cb.setDisabled(False)
-                self.infoc.setText("")
-                self.dx_HeNe = True
-                self.dx_edit.setDisabled(self.dx_HeNe)
-                self.dx = 1.0 / self.laser_wavenumber / 2.0
-            return
-        else:
-            lwn = lwn[0] if (lwn == lwn[0]).all() else ValueError()
+            self.reader = self.data.attributes['Reader']
+        except KeyError:
+            pass
+        if self.reader == 'NeaReaderGSF':
             self.dx_HeNe = False
-            self.dx_HeNe_cb.setDisabled(True)
-            self.dx_edit.setDisabled(True)
-        try:
-            udr, _ = self.data.get_column_view("Under Sampling Ratio")
-        except ValueError:
-            udr = 1
-        else:
-            udr = udr[0] if (udr == udr[0]).all() else ValueError()
+            self.dx_HeNe_cb.setDisabled(True) #??? Is possible change the label?
```

Comment:
For now, update `self.infoc` to let the user know. (like below, L620). We can fix the provided HeNe thing in a different PR since most datasets specify the value anyway.

Reply:
I show `self.dx`, but the number is large  `0.00019550342130987293`.
How can I turn them in scientific notation as you do?